src/190B20.c

```diff
@@ -1219,9 +1219,43 @@ s32 lookup_defense(DefenseTableEntry* defenseTable, Element elementKey) {
     return normalDefense;
 }
 
-INCLUDE_ASM(s32, "190B20", lookup_status_chance); // exactly (?) the same as lookup_defense
+s32 lookup_status_chance(DictionaryEntry* statusTable, Element statusKey) {
+    DictionaryEntry* row;
+    s32 normalChance = 0;
 
-INCLUDE_ASM(s32, "190B20", lookup_status_duration_mod); // exactly (?) the same as lookup_defense
+    for (row = statusTable; row->key != STATUS_END; row++, statusTable++) {
```

Comment:
This is a bit confusing.. why do we need to increment row and statusTable. Also I think this would be better with an integer iterator (i) and using indexes into statusTable

Reply:
to clarify, I think we should be able to just increment i and do statusTable[i]